src/utils.c

```diff
@@ -1832,6 +1834,87 @@ ts_is_hypercore_am(Oid amoid)
 	return amoid == hypercore_amoid;
 }
 
+/*
+ * Set reloption for relation.
+ *
+ * Most of the code is from ATExecSetRelOptions() in tablecmds.c since that
+ * function is static and we also need to do a slightly different job.
+ */
+static void
+relation_set_reloption_impl(Relation rel, List *options, LOCKMODE lockmode)
+{
+	Datum repl_val[Natts_pg_class] = { 0 };
+	bool repl_null[Natts_pg_class] = { false };
+	bool repl_repl[Natts_pg_class] = { false };
+	bool isnull;
+
+	Assert(rel->rd_rel->relkind == RELKIND_RELATION || rel->rd_rel->relkind == RELKIND_TOASTVALUE);
+
+	if (options == NIL)
+		return; /* nothing to do */
+
+	TS_DEBUG_LOG("setting reloptions for %s", RelationGetRelationName(rel));
+
+	Relation pgclass = table_open(RelationRelationId, RowExclusiveLock);
+	Oid relid = RelationGetRelid(rel);
+	HeapTuple tuple = SearchSysCache1(RELOID, ObjectIdGetDatum(relid));
+	if (!HeapTupleIsValid(tuple))
+		elog(ERROR, "cache lookup failed for relation %u", relid);
+
+	/* Get the old reloptions */
+	Datum datum = SysCacheGetAttr(RELOID, tuple, Anum_pg_class_reloptions, &isnull);
+
+	/* Generate new proposed reloptions (text array) */
+	Datum newOptions =
+		transformRelOptions(isnull ? (Datum) 0 : datum, options, NULL, NULL, false, false);
+	(void) heap_reloptions(rel->rd_rel->relkind, newOptions, true);
+
+	if (newOptions)
+		repl_val[AttrNumberGetAttrOffset(Anum_pg_class_reloptions)] = newOptions;
+	else
+		repl_null[AttrNumberGetAttrOffset(Anum_pg_class_reloptions)] = true;
+
+	repl_repl[AttrNumberGetAttrOffset(Anum_pg_class_reloptions)] = true;
+
+	HeapTuple newtuple =
+		heap_modify_tuple(tuple, RelationGetDescr(pgclass), repl_val, repl_null, repl_repl);
+
+	CatalogTupleUpdate(pgclass, &newtuple->t_self, newtuple);
+
+	/* Not sure if we need this one, but keeping it as a precaution */
+	InvokeObjectPostAlterHook(RelationRelationId, RelationGetRelid(rel), 0);
```

Comment:
I don't think we need to hook anything at this point but I'm not entirely sure as well :-(